### PR TITLE
Add basic support for CUDA native SASS assembly

### LINKED ISSUE #946: [Feature Request] Support for CUDA native assembly SASS
The PTX instructions, recently generated by CE, is NVIDIA's intermediate representation (IR). When CUDA code is compiled to PTX and run, the target GPU compiles the PTX on the fly to native assembly instructions SASS. However CUDA code can also be compiled to SASS.

SASS instructions give a better insight into performance on an actual GPU and can be dumped using either the cuobjdump or nvdisasm binaries, distributed with the CUDA toolkit, as described here
https://docs.nvidia.com/cuda/cuda-binary-utilities/index.html#differences

SASS Instruction set reference
https://docs.nvidia.com/cuda/cuda-binary-utilities/index.html#instruction-set-ref

Useful stackoverflow post
https://stackoverflow.com/questions/25476406/how-do-i-get-c-code-intermixed-with-sass-with-cuobjdump

### PR DESCRIPTION
**Pull request by benbarsdell**
- Enables "Compile to binary and disassemble the output" for CUDA source.
- Parses labels, line numbers, opcodes and disassembly.
- Fixes #946

### CODE REVIEW

**Comment by partouf** [comment]
Thank you for figuring this out. It looks good, but I would need to test it out before merge, I hope to get around that soon.

File: etc/config/cuda.amazon.properties [hunk 1]
```diff
@@ -1,7 +1,10 @@
 compilers=&nvcc:&cuclang
 defaultCompiler=nvcc92
-supportsBinary=false
+supportsBinary=true
+supportsExecute=false
+rpathFlag=-L # WAR, really need `-Xcompiler "-Wl,-rpath=<path>"`, but not required because supportsExecute=false
```
Comment (partouf) [inline etc/config/cuda.amazon.properties]: rpathFlag should be in the nvcc group, it doesn't really work for Clang

File: etc/config/cuda.amazon.properties [hunk 2]
```diff
@@ -1,7 +1,10 @@
 compilers=&nvcc:&cuclang
 defaultCompiler=nvcc92
-supportsBinary=false
+supportsBinary=true
```
Comment (partouf) [inline etc/config/cuda.amazon.properties]: I don't think binary works for cuclang, so I would suggest supportsBinary=false should be default, and group.nvcc.supportsBinary=true

File: lib/base-compiler.js [hunk 3]
```diff
@@ -153,6 +153,7 @@ class BaseCompiler {
         let args = ["-d", outputFilename, "-l", "--insn-width=16"];
         if (demangle) args = args.concat("-C");
         if (intelAsm) args = args.concat(["-M", "intel"]);
+        if (this.lang.name === 'CUDA') args = [outputFilename, "-c", "-g", "-hex"]; // For nvdisasm
```
Comment (partouf) [inline lib/base-compiler.js]: I would prefer this to be part of a specialized objdjump function within class NvccCompiler

File: etc/config/cuda.amazon.properties (hunk 1 above)
Comment (benbarsdell) [reply etc/config/cuda.amazon.properties]: Thanks for pointing this out, fixed.

File: etc/config/cuda.amazon.properties (hunk 2 above)
Comment (benbarsdell) [reply etc/config/cuda.amazon.properties]: Fixed.

File: lib/base-compiler.js (hunk 3 above)
Comment (benbarsdell) [reply lib/base-compiler.js]: Done.

**Pull request merged**